### PR TITLE
Update marketplace.xml for new PDI Step plugin PentahoGoogleSheetsPlugin

### PR DESCRIPTION
**Pull request by jfmonteil**
This is a new plugin I would like to be added to the marketplace...

Not sure this is the good procedure but let's see.

jfmonteil@gmail.com

### CODE REVIEW

File: marketplace.xml [hunk 1]
```diff
@@ -13057,4 +13057,39 @@ Uses the MarkLogic Data Movement Java SDK. Works with MarkLogic 9.0 and above on
...
+    <category>
+      <name>Input/Output</name>
+      <parent>
+        <name>PDI Plugins</name>
+      </parent>
+    </category>
+    <description>With these plugins you can read/write from/to Google Spreadsheets. This plugin uses the latest Google API's since others are deprecated March 2020</description>
+    <author>jfmonteil@gmail.com</author>
+    <author_url></author_url>
+    <author_logo>https://avatars0.githubusercontent.com/u/8617022?s=460&v=4</author_logo>
```
Comment (pamval) [inline marketplace.xml]: hi. this is an invalid xml (the & character needs to be replaced with &amp;)

Comment (jfmonteil) [reply marketplace.xml]: oups sorry you mean "&" by "`&amp;`" ?

Comment (pamval) [reply marketplace.xml]: yeah :) github formatted my message. that's exactly it

Comment (jfmonteil) [reply marketplace.xml]: updated the file, many thanks, sorry, first plugin to the marketplace !

Comment (jfmonteil) [reply marketplace.xml]: Hey Pedro, found a little bug, I have produced a 4.1 version, how do we do to proceed for the marketplace ?

Comment (pamval) [reply marketplace.xml]: change the package_url field to point to the new version and I'll merge the pr afterwards.

Comment (jfmonteil) [reply marketplace.xml]: Ok just did it

**Comment by jfmonteil** [comment]
what are the next steps ?

**Comment by pamval** [comment]
me merging the pr, which I just did. :) 

You should be good to go.

**Comment by jfmonteil** [comment]
Sorry fisrt time I publish, what should I do next ?

**Comment by pamval** [comment]
The plugin is now available in the marketplace. 
I guess you can try to download it from spoon and verify that everything is working as expected.

**Comment by jfmonteil** [comment]
Sorry, when I click on install i only have a zip file with version.xml in it installed in my plugin directory.
How can I fix that ?

**Comment by pmalves** [comment]
Where are you installing it from? Spoon, right?

**Comment by jfmonteil** [comment]
Hi Pedro, Yes, from Spoon, sorry it is my first time ;)

**Comment by jfmonteil** [comment]
I checked and the zip on my github is good, all is in it...

**Comment by jfmonteil** [comment]
Sorry guys to put pressure, any idea of what could be wrong with my XML
file ? Cheers,

**Comment by pmalves** [comment]
Have you tried installing other plugins from spoon? Do those work?

If so, try downloading one and compare the zip structure....

**Comment by jfmonteil** [comment]
Good morning Pedro,

Yes when I download from the market place, from spoon, let's say Ivytech
mongodb plugin it contains
- A lib folder (ext libs)
- A jar file (the jar of the plugin + resources such as messages)
- A version.xml file

Mine only contains
- A version.xml file.

The *marketplace.xml* submitted to the marketplace contains a link to :
<package_url>
https://github.com/jfmonteil/Pentaho-Google-Sheet-Plugin/blob/master/delivery/PentahoGoogleSheetsPlugin-4.1-distribution.zip
</package_url>
Maybe that is not a good URL because it is on Github ?

If you point to this address, the zip files contains a version.xml file and
a folder containing the plugin....
Cheers,
JFM

**Comment by pmalves** [comment]
The only thing the backend does is to download, unzip and copy to the
plugins folder... I looked at that url, and the zip has more than that, and
looks ok. (apart from the fact you seem to have more than you need on the
lib dir, but that shouldn't be an issue)

Check the logs, I can't see what would cause that to only have the
version.xml file...

**Comment by jfmonteil** [comment]
Ok I will Check,
When you download the step it works ?

Le mar. 10 mars 2020 à 12:07, Pedro Alves <notifications@github.com> a
écrit :

**Comment by jfmonteil** [comment]
My coworkers have the same problem, I will resummit the marketplace.xml
with a new marketplace URL, I think it is the root cause of  the problem...
JFM

**Comment by pamval** [comment]
Jeff, have you looked at the log files from Spoon? The process of download, unzip and install is logged there, maybe there's a hint of what might be wrong.

**Comment by jfmonteil** [comment]
spoon.log / pdi.log do not show anything :(
My coworker has the same problem.
I think it is related to the package_url,
I am trying to test a new marketplace file locally, will keep you posted

**Comment by jfmonteil** [comment]
Looks like I fixed it, the marketplace.xml was not pointing at the raw zip
but at the github page that pointed to the file, I have redone a pull
request, sorry for all the trouble and thank you.

**Pull request merged**